### PR TITLE
add route to fetch user by ID

### PR DESCRIPTION
**Pull request by isaidou**
# Pull Request Template

## Type of change

- [x] Feature
- [ ] Fix

### Summary of changes:
- Added a new `GET /api/users/:id` endpoint to retrieve a user by ID.
- Implemented the `getUserById` method in the `UserController` and `UserService`.
- Integrated proper error handling for cases where a user is not found, returning a `404 Not Found` response.
- Updated the `user.route.ts` file to include the new route.
- Enhanced Swagger documentation to include the new endpoint and its expected parameters and responses.

### Motivation:
This feature allows the frontend and other clients to fetch detailed information about a specific user using their unique identifier. It improves the usability of the API and aligns with the requirements for user management.

### Dependencies:
No additional dependencies were added for this feature.

### Tests conducted:
- [x] Verified functionality manually using Postman:
  - Tested the `GET /api/users/:id` endpoint with valid and invalid IDs.
  - Ensured proper error handling (404) for non-existent users.
- [x] Validated the Swagger documentation reflects the new endpoint correctly.
- [ ] Unit tests for `getUserById` in `UserService` are planned but not yet implemented.

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: src/apps/auth/api/user.controller.ts [hunk 1]
```diff
@@ -20,4 +20,18 @@ export class UserController extends BaseController {
...
     }
   }
+
+  async getUserById(req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+        const { id } = req.params;
+        const user = await this.userService.getUserById(id);
+        if (!user) {
+          return ApiResponse.error(res, "User not found", 404, { userId: id });
+        }
```
Comment (DaniOmer) [inline src/apps/auth/api/user.controller.ts]: il serait mieux de déplacer le check user dans le service et throw un BadRequestError si !user

File: src/apps/auth/domain/user.service.ts [hunk 2]
```diff
@@ -9,12 +9,23 @@ export class UserService {
...
       const { password, ...userWithoutPassword } = user.toObject(); 
       return userWithoutPassword as UserResponse;
     });
   }
+
+  async getUserById(id: string): Promise<UserResponse | null> {
+    const user = await this.userRepository.getById(id);
+    if (!user) {
+      return null; 
+    }
```
Comment (DaniOmer) [inline src/apps/auth/domain/user.service.ts:29]: tu pourrais throw un BadRequestError si !user

File: src/apps/auth/api/user.route.ts [hunk 3]
```diff
@@ -25,5 +25,30 @@ export default (router: Router) => {
...
+   *         description: Get user by id
+   *         content:
+   *           application/json:
+   *             schema:
+   *               $ref: '#/components/schemas/User'
+   *       404:
+   *         description: User not found
+   */
+  router.get("/:id", userController.getUserById.bind(userController));
+
```
Comment (DaniOmer) [inline src/apps/auth/api/user.route.ts:53]: Utiliser le middleware qui check la validité du format d'id `validateIdMiddleware`

**Review by johnnyhelloworld — APPROVED** [review]

File: src/apps/auth/api/user.controller.ts (hunk 1 above)
Comment (isaidou) [reply src/apps/auth/api/user.controller.ts]: fait

File: src/apps/auth/domain/user.service.ts (hunk 2 above)
Comment (isaidou) [reply src/apps/auth/domain/user.service.ts:29]: fait zebbi

File: src/apps/auth/api/user.route.ts (hunk 3 above)
Comment (isaidou) [reply src/apps/auth/api/user.route.ts:53]: eh vas y ca me soule , il y  a un antilope à la garre

**Pull request closed without merging**